tests/rwlock.rs

```diff
@@ -303,59 +303,55 @@ fn multithreaded() {
 fn with_read_err() {
     let mtx = RwLock::<i32>::new(-5);
     let r = current_thread::block_on_all(lazy(|| {
-        let fut = mtx.with_read(|guard| {
+        mtx.with_read(|guard| {
             if *guard > 0 {
                 Ok(*guard)
             } else {
                 Err("Whoops!")
             }
-        });
-        fut.map(|r| assert_eq!(r, Err("Whoops!")))
+        })
     }));
-    assert!(r.is_ok());
+    assert_eq!(r, Err("Whoops!"));
 }
 
 #[cfg(feature = "tokio")]
 #[test]
 fn with_read_ok() {
     let mtx = RwLock::<i32>::new(5);
     let r = current_thread::block_on_all(lazy(|| {
-        let fut = mtx.with_read(|guard| {
+        mtx.with_read(|guard| {
             Ok(*guard) as Result<i32, ()>
-        });
-        fut.map(|r| assert_eq!(r, Ok(5)))
+        })
     }));
-    assert!(r.is_ok());
+    assert_eq!(r, Ok(5));
 }
 
 #[cfg(feature = "tokio")]
 #[test]
 fn with_write_err() {
     let mtx = RwLock::<i32>::new(-5);
     let r = current_thread::block_on_all(lazy(|| {
-        let fut = mtx.with_write(|mut guard| {
+        mtx.with_write(|mut guard| {
             if *guard > 0 {
                 *guard -= 1;
                 Ok(())
             } else {
                 Err("Whoops!")
             }
-        });
-        fut.map(|r| assert_eq!(r, Err("Whoops!")))
+        })
     }));
-    assert!(r.is_ok());
+    assert_eq!(r, Err("Whoops!"));
 }
 
 #[cfg(feature = "tokio")]
 #[test]
 fn with_write_ok() {
     let mtx = RwLock::<i32>::new(5);
-    let r = current_thread::block_on_all(lazy(|| {
-        let fut = mtx.with_write(|mut guard| {
+    let _ = current_thread::block_on_all(lazy(|| {
```

Comment:
`let _ =` makes it look like you're doing something wrong.  Is that there to suppress a warning about an unused `Result`?  How about calling `.unwrap()` instead?  You can freely ignore the result of `.unwrap()`.